app/src/main/java/uk/gov/onelogin/wallet/DeleteWalletDataUseCase.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.onelogin.wallet
+
+import androidx.fragment.app.FragmentActivity
+import javax.inject.Inject
+import uk.gov.android.wallet.sdk.WalletSdk
+
+fun interface DeleteWalletDataUseCase {
+    suspend operator fun invoke(activityFragment: FragmentActivity)
```

Comment:
Please create a named function that represents what is going on (clearWalletData?) and don't override the invoke operator. I understand that in our current usecase the invoke operator is used, but going forward we will be moving towards clean coding, of which this violates a significant number of principles (none of which are hard rules, but are there to point the correct direction), including:

- Naming convention, Verbs for actions: as invoke is not required at the calling point, suddenly you have (or can have) a noun (class name) performing an action instead of a verb (function name)
- Naming convention, functions named for action to be performed: Invoke tells us what is happening with the class, but gives no indication to the calling site about the action that is actually being performed, as it is overwritten
- Throws exception in constructor: It is convention (in all OO languages) to forbid exceptions being thrown in constructors, which in Kotlin extends to init blocks and invoke calls, as these are called at object construction
- Idiomatic coding: Any programmer familiar with OO should be able to read the code we write (especially us in the future), overriding the invoke operator is extremely specific to Kotlin, and causes confusion as you suddenly have object construction used as a function call `val objectA = SomeObject()`, should be reserved for actual object creation and not performing 'hidden' actions when an action is not expected

There are others, but that is what I have off the top of my head, without having to go get a text book.
It is one of those things that is very 'clever', but we don't want clever, we want dumb / simple / easy

Reply:
Pulling out discussion on "overriding the invoke" to a tech debt ticket, so it can be discussed in more detail.